Refactor multilayer drawing function and add headers to convert module

nwlandry opened the pull request:
This adds the ability for users to specify the colors/sizes of every node/edge in the multilayer drawing function.

nwlandry commented:
Happy for any feedback @thomasrobiglio!!

thomasrobiglio commented:
@nwlandry let me know if this is still a draft or you want me to review it (it looks like the function is ready we just need to twitch the tests)... I tried it very quickly today and it seems very nice and much more coherent with the rest of the drawing functions of XGI than my initial attempt 😄, thank you!

nwlandry commented:
Hi @thomasrobiglio, yes sorry about this - let me draft it and undraft when I am done updating the docs, tests, etc. I had a great foundation to work from, so thanks :)

nwlandry commented on xgi/drawing/draw_utils.py:
```diff
@@ -0,0 +1,254 @@
+
+
+def _draw_init(H, ax, pos):
+
```
Done! Thanks for the suggestion!

nwlandry commented:
I added a `plane_color` option with the default as grey. Because I rebased, it looks like I reorganized the drawing module, but that is already on the main branch. Thanks for the review!!
